man/froll.Rd

```diff
@@ -83,37 +69,68 @@ frollapply(x, n, FUN, \dots, fill=NA, align=c("right", "left", "center"))
   to compute a rolling average is inefficient. It is also always single-threaded
   because there is no thread-safe API to R's C \code{eval}. Nevertheless we've
   seen the computation speed up vis-a-vis versions implemented in base R.
+
+  Setting \code{options(datatable.verbose=TRUE)} will display various
+  information about how rolling function processed. It will not print
+  information in a real-time but only at the end of the processing.
 }
 \value{
-  A list except when the input is a \code{vector} and
-  \code{length(n)==1} in which case a \code{vector} is returned.
+  A list except when the input is a \code{vector} and \code{length(n)==1}, in which case a \code{vector} is returned, for convenience. Thus, rolling functions can be used conveniently within \code{data.table} syntax.
 }
 \note{
-  Users coming from most popular package for rolling functions
-  \code{zoo} might expect following differences in \code{data.table}
-  implementation.
+  Be aware that rolling functions operates on the physical order of input. If the intent is to roll values in a vector by a logical window, for example an hour, or a day, then one has to ensure that there are no gaps in input. For details see \href{https://github.com/Rdatatable/data.table/issues/3241}{issue #3241}.
+}
+\section{\code{hasNA} argument}{
+  \code{hasNA} can be used to speed up processing in cases when it is known if \code{x} contains \code{NA} (or \code{NaN}).
   \itemize{
-    \item rolling function will always return result of the same length as input.
-    \item \code{fill} defaults to \code{NA}. 
+    \item Default \code{hasNA=NA} will use faster NA inaware implementation, but when NAs are detected it will re-run NA aware implementation.
+    \item \code{hasNA=TRUE} will use NA aware implementation straightaway.
+    \item \code{hasNA=FALSE} will use faster NA inaware implementation. Then depending on the rolling function it will either:
+    \itemize{
+      \item (\emph{mean, sum}) detect NAs, raise warning, re-run NA aware.
+      \item (\emph{max}) not detect NAs and may silently produce incorrect answer.
+    }
+    Therefore \code{hasNA=FALSE} should be used with care.
+  }
+}
+\section{Implementation}{
+  \itemize{
+    \item \code{algo="fast"} uses \emph{"on-line"} algorithm, and all of \code{NaN, +Inf, -Inf} (Inf are to be changed!!) are treated as \code{NA}. Not all functions have \emph{fast} implementation available. As of now \emph{max} and \code{adaptive=TRUE} does not have, therefore it will automatically fall back to \emph{exact} implementation. \code{datatable.verbose} option can be used to check that.
```

Comment:
```suggestion
    \item \code{algo="fast"} uses \emph{"on-line"} algorithm, and all of \code{NaN, +Inf, -Inf} (Inf is to be changed!!) are treated as \code{NA}. Not all functions have \emph{fast} implementation available. As of now, \emph{max} and \code{adaptive=TRUE} does not have, therefore it will automatically fall back to \emph{exact} implementation. \code{datatable.verbose} option can be used to check that.
```